flytekit/types/structured/basic_dfs.py

```diff
@@ -75,10 +79,13 @@ def encode(
     ) -> literals.StructuredDataset:
         path = typing.cast(str, structured_dataset.uri) or ctx.file_access.get_random_remote_path()
         df = structured_dataset.dataframe
-        local_dir = ctx.file_access.get_random_local_directory()
-        local_path = os.path.join(local_dir, f"{0:05}")
-        pq.write_table(df, local_path)
-        ctx.file_access.upload_directory(local_dir, path)
+        if flytekit.is_fsspec_io():
+            pq.write_table(df, path)
+        else:
+            local_dir = ctx.file_access.get_random_local_directory()
+            local_path = os.path.join(local_dir, f"{0:05}")
+            pq.write_table(df, local_path)
```

Comment:
I'm not sure if we can do this. 
Do I need to add a new flytekit-plugin to override the default encoder?